internal/command/deploy/strategy_rolling.go

```diff
@@ -0,0 +1,171 @@
+		}
+	}
+	return nil
+}
+
+func (md *machineDeployment) updateMachineByReplace(ctx context.Context, e *machineUpdateEntry, indexStr string) error {
+	lm := e.leasableMachine
+	// If machine requires replacement, destroy old machine and launch a new one
+	// This can be the case for machines that changes its volumes.
+	fmt.Fprintf(md.io.ErrOut, "  %s Replacing %s by new machine\n", indexStr, md.colorize.Bold(lm.FormattedMachineId()))
+	if err := lm.Destroy(ctx, true); err != nil {
+		return err
+	}
+
+	// Acquire a lease on the new machine to ensure external factors can't stop or update it
+	// while we wait for its state and/or health checks
+	e.launchInput.LeaseTTL = int(md.waitTimeout.Seconds())
+
+	newMachineRaw, err := md.flapsClient.Launch(ctx, *e.launchInput)
+	if err != nil {
+		return err
+	}
+
+	lm = machine.NewLeasableMachine(md.flapsClient, md.io, newMachineRaw)
+	fmt.Fprintf(md.io.ErrOut, "  %s Created machine %s\n", indexStr, md.colorize.Bold(lm.FormattedMachineId()))
+	defer lm.ReleaseLease(ctx)
+	e.leasableMachine = lm
+	return nil
+}
+
+func (md *machineDeployment) updateMachineInPlace(ctx context.Context, e *machineUpdateEntry, indexStr string) error {
+	lm := e.leasableMachine
+	fmt.Fprintf(md.io.ErrOut, "  %s Updating %s\n", indexStr, md.colorize.Bold(lm.FormattedMachineId()))
+	if err := lm.Update(ctx, *e.launchInput); err != nil {
+		if md.strategy != "immediate" {
+			return err
+		}
+		fmt.Fprintf(md.io.ErrOut, "Continuing after error: %s\n", err)
+	}
+	return nil
+}
+
+func (md *machineDeployment) waitForMachine(ctx context.Context, lm machine.LeasableMachine, inBatch bool, indexStr string) error {
+	if md.strategy == "immediate" {
+		return nil
+	}
+
+	// Don't wait for Standby machines, they are updated but not started
+	if len(lm.Machine().Config.Standbys) > 0 {
+		md.logClearLinesAbove(1)
+		fmt.Fprintf(md.io.ErrOut, "  %s Machine %s update finished: %s\n",
+			indexStr,
+			md.colorize.Bold(lm.FormattedMachineId()),
+			md.colorize.Green("success"),
+		)
+		return nil
+	}
+
+	if !md.skipHealthChecks {
+		if err := lm.WaitForState(ctx, api.MachineStateStarted, md.waitTimeout, indexStr, false); err != nil {
+			err = suggestChangeWaitTimeout(err, "wait-timeout")
+			return err
+		}
+	}
+
+	if err := md.doSmokeChecks(ctx, lm, indexStr); err != nil {
+		return err
+	}
+
+	if !md.skipHealthChecks {
+		if err := lm.WaitForHealthchecksToPass(ctx, md.waitTimeout, indexStr); err != nil {
+			md.warnAboutIncorrectListenAddress(ctx, lm)
+			err = suggestChangeWaitTimeout(err, "wait-timeout")
+			return err
+		}
+		// FIXME: combine this wait with the wait for start as one update line (or two per in noninteractive case)
+		md.logClearLinesAbove(1)
+		fmt.Fprintf(md.io.ErrOut, "  %s Machine %s update finished: %s\n",
+			indexStr,
+			md.colorize.Bold(lm.FormattedMachineId()),
+			md.colorize.Green("success"),
+		)
+		if inBatch {
+			fmt.Fprint(md.io.ErrOut, "\n")
+		}
+	}
+
+	md.warnAboutIncorrectListenAddress(ctx, lm)
```

Comment:
This function may modify `md.listenAddressChecked` map. Should the map now be protected with a mutex?